### PR TITLE
rEmoved inplace probe loading

### PR DESCRIPTION
**Pull request by colehurwitz**
@alejoe91

### CODE REVIEW

**Comment by colehurwitz** [comment]
This seems like it will still work and passes all tests. Let me know what you think please.

File: spikeextractors/extraction_tools.py [hunk 1]
```diff
@@ -86,14 +85,10 @@ def load_probe_file(recording, probe_file, channel_map=None, channel_groups=None
...
-
-            if list(ordered_channels) == recording.get_channel_ids():
-                subrecording = recording
-            else:
-                if not np.all([chan in recording.get_channel_ids() for chan in ordered_channels]) and verbose:
-                    print('Some channel in PRB file are not in original recording')
-                present_ordered_channels = [chan for chan in ordered_channels if chan in recording.get_channel_ids()]
-                subrecording = SubRecordingExtractor(recording, channel_ids=present_ordered_channels)
+            if not np.all([chan in recording.get_channel_ids() for chan in ordered_channels]) and verbose:
+                print('Some channel in PRB file are not in original recording')
```
Comment (samuelgarcia) [inline spikeextractors/extraction_tools.py]: I would raise an error instead no ?

Comment (alejoe91) [reply spikeextractors/extraction_tools.py]: No this is perfectly fine. Some channels could be ref and users might want to exclude them from the probe file..I think we discussed about it already!

**Comment by alejoe91** [comment]
@colehurwitz looks good to me

**Comment by samuelgarcia** [comment]
@colehurwitz @alejoe91 
I have just realise that this PR have broken a important feature for me.

In some sorters (KS, KS2, klusta, TDC) to avoid a heavy copy each time I tested in sorter if the input was an BinDatRecordingExtractor with **isinstance(recording, se.BinDatRecordingExtractor)**.
When it was the case then no local copy and instead the filename was given to the sorter.
It was very convinient in ComparisonStudy because the input is always BinDatRecordingExtractor and so no copy was perform at all.

Now with this **not inplace load probe**, the recording is not anymore a BinDatRecordingExtractor but indeed a SubRecordingExtractor.
So the feature is broken for me and it was quite important.
How could we find a solution for this.

Copy everything each time for each sorter is so painfull we I work on benchmark. How could we fix that ?

**Comment by alejoe91** [comment]
Mmm good point @samuelgarcia...I don't know what the best would be. Maybe for subrecording extractor to store the type of the parent extractor? Unfortunately I think that loading the probe file must return a subrecording, otherwise reordering/skipping channels gets complicated

**Comment by samuelgarcia** [comment]
Totally agree the load_probe must not be in place and have to return subrecording.
But the need of not copy everything is really important.

I think a quick and dirty fix could be to make a load_probe_in_place that would be used only in the context of ComparisonStudy.
In comparison study, the recording is copied (and shared) inside "raw_files" folder. In that case the channel map is complete and the PRB is one group always. It is a simple case.

So in that specific case we could have a load_probe_file_inplace that would be call by ComparisonStudy when loading the PRB. This load_probe_file_inplace could be in studytools and not in spikeextractor to avoid mistake.

What do you think ?

**Pull request merged**